core/src/main/scala/com/salesforce/op/stages/impl/selector/ModelSelectorFactory.scala

```diff
@@ -58,25 +66,33 @@ private[op] trait ModelSelectorFactory {
    * @param trainTestEvaluators evaluation to do on data
    * @param modelTypesToUse list of models to use
    * @param modelsAndParameters sequence of models and parameters to explore
+   * @param modelDefaults default model types and model parameters for problem type
    * @return model selector with these settings
    */
   protected def selector(
     validator: OpValidator[ModelType, EstimatorType],
     splitter: Option[Splitter],
     trainTestEvaluators: Seq[OpEvaluatorBase[_ <: EvaluationMetrics]],
     modelTypesToUse: Seq[ModelsToTry],
-    modelsAndParameters: Seq[(EstimatorType, Array[ParamMap])]
+    modelsAndParameters: Seq[(EstimatorType, Array[ParamMap])],
+    modelDefaults: ModelDefaults[_ <: ModelsToTry]
   ): ModelSelector[ModelType, EstimatorType] = {
-    val modelStrings = modelTypesToUse.map(_.entryName)
-    val modelsToUse =
-    // if no models are specified use the defaults and filter by the named models to use
-      if (modelsAndParameters.isEmpty) defaultModelsAndParams
-        .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
-      // if models to use has been specified and the models have been specified filter the models by the names
-      else if (modelTypesToUse != modelNames) modelsAndParameters
-        .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+    val modelTypeNames = modelTypesToUse.map(_.entryName).toSet
+    val modelsToUse = {
+      // if no models are specified use the defaults and filter by the named models to use
+      if (modelsAndParameters.isEmpty) {
+        modelDefaults.modelsAndParams.filter { case (e, p) => modelTypeNames(e.getClass.getSimpleName) }
+      }
+      // if models to use has been specified and the models have been specified - filter the models by the names
+      else if (
+        modelTypesToUse.distinct.sortBy(_.entryName) != modelDefaults.modelTypesToUse.distinct.sortBy(_.entryName)
```

Comment:
Why not simply use entire modelsAndParameters?

Reply:
well, because it would yield incorrect results: `List(1,2) == List(2,1) // false`. That the reason I am dedupping and sorting here.

Reply:
I was thinking whether it made more sense to ignore modelTypes and modelDefaults when modelsAndParameters was not empty.